Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h

```diff
@@ -2363,7 +2363,7 @@ class Property_map
     template<class I>
     std::vector<std::string> properties() const
     {
-      return Property_selector<I>(this)().properties();
+      return Property_selector<I>(const_cast<Self*>(this))().properties();
     }
 
```

Comment:
you are right. Additionally it is not critical unless you have tiny meshes with many properties and many calls of the function.